File: 20220817/모범답안/20220817_07.js/CountMaximumWords.py
from functools import reduce


def solution(arr):
    """
    :param arr: str[]
    :return: int
    """

    counts = list(map(lambda s: countWords(s), arr))
    return reduce(lambda pre, cur: max(pre, cur), counts, 0)


def countWords(s):
    return len(s.split(' '))

File: 20220817/모범답안/20220817_07.js/CountMaximumWords.js
/**
 * @param arr {string[]}
 * @returns {number}
 */
function solution(arr) {
  return arr.map(s => countWords(s))
    .reduce((pre, cur) => Math.max(pre, cur), 0)
}

function countWords(s) {
  return s.split(' ').length
}

solution